### PR TITLE
Add timeout to hffs requests

### PR DESCRIPTION
**Pull request by lhoestq**
I think it causes the fineweb job to hang, and then kube thinks the pod is doing nothing and can kill it

cc @polinaeterna 

also cc @Wauplin just FYI

### CODE REVIEW

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 1]
```diff
@@ -1261,7 +1267,8 @@ def compute_config_parquet_and_info_response(
             parquet_operations = copy_parquet_files(builder)
             logging.info(f"{len(parquet_operations)} parquet files to copy for {dataset=} {config=}.")
             validate = ParquetFileValidator(max_row_group_byte_size=max_row_group_byte_size_for_copy).validate
-            fill_builder_info(builder, hf_endpoint=hf_endpoint, hf_token=hf_token, validate=validate)
+            with patch("huggingface_hub.hf_file_system.http_backoff", http_backoff_with_timeout):
```
Comment (Wauplin) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:1270]: Feels quite hacky :confused: 
No solution to do that differently? What's the default timeout and what the value you would prefer?

Comment (lhoestq) [reply services/worker/src/worker/job_runners/config/parquet_and_info.py:1270]: Default is None (can hang indefinitely). In `datasets` we use 100 by default.

I'm using this hack right now to validate this issue, we can see later for a proper fix on the hfh side

**Comment by lhoestq** [comment]
It worked thanks to this fix :)

and indeed in the log it remedied some time outs

```
'(ReadTimeoutError("HTTPSConnectionPool(host='hf.co', port=443): Read timed out. (read timeout=10)"), '(Request ID: 7fe2be5e-1645-4181-9715-0eff6a748b57)')' thrown while requesting GET https://huggingface.co/datasets/HuggingFaceFW/fineweb/resolve/fa3ba60176ae9a23452ad2d81cce141bdcd635f0/data/CC-MAIN-2020-50/000_00031.parquet
WARNING: 2024-04-25 21:08:45,002 - huggingface_hub.utils._http - '(ReadTimeoutError("HTTPSConnectionPool(host='hf.co', port=443): Read timed out. (read timeout=10)"), '(Request ID: 7fe2be5e-1645-4181-9715-0eff6a748b57)')' thrown while requesting GET https://huggingface.co/datasets/HuggingFaceFW/fineweb/resolve/fa3ba60176ae9a23452ad2d81cce141bdcd635f0/data/CC-MAIN-2020-50/000_00031.parquet
Retrying in 1s [Retry 1/5].
WARNING: 2024-04-25 21:08:45,002 - huggingface_hub.utils._http - Retrying in 1s [Retry 1/5].
```

**Comment by severo** [comment]
Do you have an estimate of the number of errors that might be due to the lack of this timeout? Hopefully it enables a lot of new datasets

**Comment by Wauplin** [comment]
I opened https://github.com/huggingface/huggingface_hub/pull/2253 to properly handle this in `huggingface_hub`. Actually we already have default timeout values for `hf_hub_download` so it'd normal to use them in hf filesystem as well.

**Comment by lhoestq** [comment]
> Do you have an estimate of the number of errors that might be due to the lack of this timeout? Hopefully it enables a lot of new datasets

Not sure. If the frequency of those errors is one every 20k files it should be quite rare in regular repos.
It must have happened for other big datasets like the-stack or the-stack-v2 for example

**Pull request merged**